myuw/dao/mailman.py

```diff
@@ -0,0 +1,233 @@
+        section_id = get_section_id(url)
+        secondaries_section_ids.append(section_id)
+        thread = SingleListThread(primary_section.curriculum_abbr,
+                                  primary_section.course_number,
+                                  section_id,
+                                  primary_section.term.quarter,
+                                  primary_section.term.year)
+        thread.start()
+        list_threads[section_id] = thread
+
+    for section_id in secondaries_section_ids:
+        thread = list_threads[section_id]
+        thread.join()
+        if thread.exception is None:
+            secondaries.append(thread.response)
+        else:
+            logger.error("get_single_course_list(%s,%s,%s,%s,%s)==>%s " %
+                         (primary_section.curriculum_abbr,
+                          primary_section.course_number,
+                          section_id,
+                          primary_section.term.quarter,
+                          primary_section.term.year,
+                          thread.exception))
+    return secondaries
+
+
+def get_section_email_lists(section, include_secondaries_in_primary):
+    """
+    @param section: a valid sws.Section object
+    """
+    is_primary_section = section.is_primary_section
+    json_data = {
+        "course_abbr": section.curriculum_abbr,
+        "course_number": section.course_number,
+        "section_id": section.section_id,
+        "is_primary": is_primary_section,
+        }
+    json_data["section_list"] = get_single_section_list(section)
+
+    if is_primary_section:
+        if include_secondaries_in_primary and\
+                len(section.linked_section_urls) > 0:
+            json_data["secondary_lists"] =\
+                get_all_secondary_section_lists(section)
+
+        json_data["has_multi_secondaries"] =\
+            (len(section.linked_section_urls) > 1)
+
+        if json_data["has_multi_secondaries"]:
+            json_data["secondary_combined_list"] =\
+                get_section_secondary_combined_list(section)
+    return json_data
+
+
+def get_email_lists_by_term(term, include_secondaries_in_primary=True):
+    """
+    @return json of the corresponding email lists
+    associated with the current instructor in the given term
+    """
+    json_data = {
+        "year": term.year,
+        "quarter": term.quarter,
+        }
+
+    schedule = get_instructor_schedule_by_term(term)
+
+    if len(schedule.sections) >= 0:
+        json_data["instructor_term_list"] =\
+            get_instructor_term_list(get_netid_of_current_user(),
+                                     term.quarter,
+                                     term.year)
+
+    json_data["email_lists"] = []
+    if len(schedule.sections) == 1:
+        json_data["email_lists"].append(
+            get_section_email_lists(schedule.sections[0],
+                                    include_secondaries_in_primary))
+
+    if len(schedule.sections) > 1:
```

Comment:
That's a good question.  While we're waiting to see how all of the section related code needs to change, I'd rather err on the side of failing intentionally, rather than trying to do too much.  

I guess I'm not really viewing this as making a decision, but more like building in an escape valve